src/main/java/ch/njol/skript/expressions/ExprNumbers.java

```diff
@@ -106,34 +127,63 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			f = temp;
 		}
 		final Number starting = s, finish = f;
-		return new Iterator<Number>() {
-			double i = integer ? Math.ceil(starting.doubleValue()) : starting.doubleValue(), max = integer ? Math.floor(finish.doubleValue()) : finish.doubleValue();
-			
-			@Override
-			public boolean hasNext() {
-				return i <= max;
-			}
+		if (mode < 2) {
+			return new Iterator<Number>() {
+				double i = mode == 1 ? Math.ceil(starting.doubleValue()) : starting.doubleValue();
+				double max = mode == 1 ? Math.floor(finish.doubleValue()) : finish.doubleValue();
+				
+				
+				@Override
+				public boolean hasNext() {
+					return i <= max;
+				}
+				
+				@Override
+				public Number next() {
+					if (!hasNext())
+						throw new NoSuchElementException();
+					if (mode == 1)
+						return (long) (reverse ? max-- : i++);
+					else
+						return reverse ? max-- : i++;
+				}
+			};
+		} else {
+			return new Iterator<Number>() {
+				final double start = starting.doubleValue();
+				final double max = finish.doubleValue();
+				final String[] split = starting.toString().split("\\.");
+				final int precision = split.length > 0 ? split[1].length() : 0;
 			
-			@Override
-			public Number next() {
-				if (!hasNext())
-					throw new NoSuchElementException();
-				if (integer)
-					return (long) (reverse ? max-- : i++);
-				else
-					return reverse ? max-- : i++;
-			}
-		};
+				final double multiplier = Math.pow(10, precision);
+				int current = reverse ? (int) (max * multiplier - 1) : (int) (start * multiplier);
+				
+				@Override
+				public boolean hasNext() {
+					return reverse ? current > start : current < max * multiplier;
+				}
+				
+				@Override
+				public Number next() {
+					if (!hasNext())
+						throw new NoSuchElementException();
+					double value = start + current / multiplier;
+					current = reverse ? current - 1 : current + 1;
+					return value;
+				}
+			};
+		}
 	}
 	
 	@Override
 	public String toString(final @Nullable Event e, final boolean debug) {

```

Comment:
Not that this is on you, but since you're already editing this expression can you move toString to the bottom of the file?

Reply:
Done! :)